yt_dlp/extractor/ximalaya.py

```diff
@@ -71,8 +72,157 @@ class XimalayaIE(XimalayaBaseIE):
                 'like_count': int,
             },
         },
+        {
+            # VIP-restricted audio
+            'url': 'https://www.ximalaya.com/sound/562111701',
+            'only_matching': True,
+        },
     ]
 
+    @staticmethod
+    def _decrypt_filename(audio_info):
+        seed = float_or_none(audio_info['seed'])
+        file_id = audio_info['fileId']
+        cgstr = ''
+        key = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ/\\:._-1234567890'
+        for _ in range(len(key)):
+            j = 211 * seed + 30031
+            seed = float_or_none(int_or_none(j) % 65536)
+            ran = seed / float_or_none(65536)
+            r = int_or_none(ran * float_or_none(len(key)))
```

Comment:
Should be identical to this, no?
```suggestion
            r = int(seed / 65536 * len(key))
```